caravel/views.py

```diff
@@ -561,13 +557,28 @@ def save(self, args, slc):
         slc.datasource_type = datasource_type
         slc.slice_name = slice_name
 
-        if action == "save":
-            session.add(slc)
-        elif action == "overwrite":
-            session.merge(slc)
+        if action == 'save':
+            self.save_slice(slc)
+        elif action == 'overwrite':
+            self.overwrite_slice(slc)
+
+        return redirect(slc.slice_url)
+
+    @has_access
```

Comment:
Oh I think this will create new view related permissions, where we should instead use the already defined model permissions using `if self.appbuilder.sm.has_access('can_add', 'Slices'):` and 
`if self.appbuilder.sm.has_access('can_edit', 'Slices'):`